### PR TITLE
Enable handling of issue comments on GitLab

### LINKED ISSUE #983: /packit propose-downstream should react to GitLab Issue/MR comments
```
[2021-02-17 08:40:44,591: DEBUG/ForkPoolWorker-1] We did not find any handler for a following event:
<class 'packit_service.service.events.IssueCommentGitlabEvent'>
```

It would be great if we were able to `propose-downstream` after an MR is merged

### PR DESCRIPTION
**Pull request by mfocko**
Fixes #983

Signed-off-by: Matej Focko <mfocko@redhat.com>

### CODE REVIEW

**Comment by mfocko** [comment]
- [ ] gotta refactor the `issue_comment` tests, some Friday work
- [ ] I don't like this: [packit_service/service/events.py:700](https://github.com/packit/packit-service/pull/985/files#diff-7c4f508e377bb4c8cd4f0b1b6d523d5261f753b4f3e31fdeb4d2b50d40f92a9eR700)

File: packit_service/service/events.py [hunk 1]
```diff
@@ -697,18 +697,33 @@ def __init__(
     ):
         super().__init__(project_url=project_url)
         self.action = action
+        # issue_id = ID of the specific comment
         self.issue_id = issue_id
```
Comment (TomasTomecek) [inline packit_service/service/events.py:701]: so how about renaming it to issue_comment_id? :)

File: packit_service/service/events.py [hunk 2]
```diff
@@ -697,18 +697,33 @@ def __init__(
...
+    @property
+    def tag_name(self):
+        if not self._tag_name:
+            releases = self.project.get_releases()
+            self._tag_name = releases[0].tag_name if releases else ""
+        return self._tag_name
+
+    @property
+    def commit_sha(self):
+        return self.tag_name
```
Comment (TomasTomecek) [inline packit_service/service/events.py:720]: these two are indeed pretty cryptic, if you refactor them, comments/docstrings would be greatly appreciated!

**Review by TomasTomecek — COMMENTED** [review]
looks good, I'd try it :)

File: packit_service/service/events.py (hunk 2 above)
Comment (mfocko) [reply packit_service/service/events.py:720]: when doing documentation for `ogr` I noticed there's `get_latest_release` method, so I'll have a look into it and check how it behaves with no releases

File: packit_service/service/events.py (hunk 1 above)
Comment (mfocko) [reply packit_service/service/events.py:701]: it's used in database, damn... will try to do something about it with rebase

i'm more for merging, and refactor + fixes if there is some problem (it's GitLab and it's special :upside_down_face:)

File: packit_service/service/events.py (hunk 2 above)
Comment (mfocko) [reply packit_service/service/events.py:719]: filed bug report, will fix tomorrow

File: packit_service/service/events.py (hunk 1 above)
Comment (mfocko) [reply packit_service/service/events.py]: should be resolved

File: packit_service/service/events.py [hunk 3]
```diff
@@ -229,7 +229,10 @@ def db_trigger(self) -> Optional[AbstractTriggerDbType]:
                     project_url=self.project_url,
                     commit_hash=self.commit_sha,
                 )
-            elif self.event_type in {IssueCommentEvent.__name__}:
+            elif self.event_type in {
+                IssueCommentEvent.__name__,
+                IssueCommentGitlabEvent.__name__,
+            }:
```
Comment (lachmanfrantisek) [inline packit_service/service/events.py:235]: :+1:

File: packit_service/worker/handlers/github_handlers.py [hunk 4]
```diff
@@ -147,6 +148,7 @@ def run(self) -> TaskResults:
 @run_for_comment(command="propose-update")  # deprecated
 @reacts_to(event=ReleaseEvent)
 @reacts_to(event=IssueCommentEvent)
+@reacts_to(event=IssueCommentGitlabEvent)
```
Comment (lachmanfrantisek) [inline packit_service/worker/handlers/github_handlers.py:151]: :+1:

**Review by lachmanfrantisek — APPROVED** [review]

**Pull request merged**